src/lib/app/mainapplication.cpp

```diff
@@ -403,17 +403,17 @@ MainApplication::AfterLaunch MainApplication::afterLaunch() const
     return static_cast<AfterLaunch>(Settings().value(QSL("Web-URL-Settings/afterLaunch"), RestoreSession).toInt());
 }
 
-bool MainApplication::restoreSession(BrowserWindow* window, RestoreData restoreData)
+void MainApplication::openSession(BrowserWindow* window, RestoreData &restoreData)
 {
-    if (m_isPrivate || restoreData.isEmpty()) {
-        return false;
-    }
-
-    m_isRestoring = true;
     setOverrideCursor(Qt::BusyCursor);
 
+    if (!window)
+        window = createWindow(Qz::BW_OtherRestoredWindow);
+
     window->setUpdatesEnabled(false);
-    window->tabWidget()->closeRecoveryTab();
+
+    if (m_isRestoring)
+        window->tabWidget()->closeRecoveryTab();
```

Comment:
Shouldn't this be called everytime? If there is no recovery tab it does nothing anyway.